### DISCUSSION TITLE
A method to check if a follower node is alive or unresponsive

### DISCUSSION THREAD

**Discussion by aley1** [dotnet/dotNext#137] [Q&A]
@Sakno, 

Is there a way to detect or check if a follower node is alive (or unresponsive) through DotNext Tcp library?
e.g. to check if Node is available before sending a message to the node.

**Comment by sakno** [comment]
[IClusterMember.Status](https://dotnet.github.io/dotNext/api/DotNext.Net.Cluster.IClusterMember.html#DotNext_Net_Cluster_IClusterMember_Status). However, it is not practical for the purpose you mentioned. Due to async nature, the member may become unresponsive immediately after status check but before message send:
```csharp
if (member.IsAvailable) // true
  // member becomes unavailable
  member.SendAsync(); // exception
```

**Comment by aley1** [comment]
I've tried checking RaftClusterMember.Status of each member in the raft cluster and the status is always 'unknown'. 
example:
if (IsLeader())
                {
                    foreach (RaftClusterMember member in _raftCluster.Members)
                    {
                         Console.WriteLine($"Cluster member status: {member.Status}");
                    }
                }
}
Could it be a bug?

**Reply by sakno** [reply]
Seems like a bug, for HTTP it works as expected. ~~Could you please create an issue?~~ #139
